### PR TITLE
Bump GitHub Action versions

### PR DESCRIPTION
**Pull request by domesticmouse**
## Pre-launch Checklist

- [x] I read the [Effective Dart: Style] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-devrel channel on [Discord].

[Effective Dart: Style]: https://dart.dev/guides/language/effective-dart/style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by domesticmouse** [comment]
Hey @jmagman, any ideas on how to fix the macOS compilation on beta? I've tried installing rosetta, but it looks like I'm holding it wrong.

**Comment by domesticmouse** [comment]
Hey @mit-mit, looks like we have a CI issue with the Dart included in Flutter 2.13.0-0 on GitHub actions for macOS only.

Context: https://github.com/flutter/codelabs/runs/6050738007?check_suite_focus=true#step:5:10

**Comment by mit-mit** [comment]
I tought we were adding ARM64 support, so do we still need Rosetta enabled? cc @cbracken

**Comment by jmagman** [comment]
> I tought we were adding ARM64 support, so do we still need Rosetta enabled? cc @cbracken

Yes, flutter tooling still requires Rosetta, in particular when a macOS app builds x64 in release mode on arm64 it uses the x64 version of `gen_snapshot`.  

However that `Run dart --version ...: Bad CPU type in executable`error means that the x64 version of dart is being downloaded instead of the arm64 version.

Screen Shot 2022-04-19 at 1 15 13 PM

Where is that cache coming from?

**Comment by domesticmouse** [comment]
Hey @jmagman,

If I attempt to enable rosetta on GitHub Actions without an architecture guard it dies, because GitHub Actions is (currently) running on Intel CPUs. I think this PR is worth landing for the future when GitHub gets M1 (or M2) based build bots, but as you can see, adding Rosetta isn't fixing this build failure. So, we need another fix. Halp?

**Comment by jmagman** [comment]
> GitHub Actions is (currently) running on Intel CPUs

Oh weird, then maybe it's the opposite, the version of dart being used is arm instead of x64?

**Comment by mit-mit** [comment]
From the CI it's failing with Flutter `flutter/2.13.0-0.1.pre-beta/`. Looking at https://docs.flutter.dev/development/tools/sdk/releases?tab=macos, that is the first ever release that has both x64 and arm, so I wonder if this is a bug in setup-flutter, that it picks arm when it should be picking x64?

**Comment by domesticmouse** [comment]
@mit-mit whom do we hassle to set up a test to start debugging `setup-flutter`

**Comment by mit-mit** [comment]
I think an issue filed here? https://github.com/subosito/flutter-action/issues

**Comment by jmagman** [comment]
They merged this 2 days ago https://github.com/subosito/flutter-action/pull/147

**Comment by jmagman** [comment]
So this would need to be updated
https://github.com/flutter/codelabs/blob/8a38fed0abb1bfd6b91a89aca9356285c8a3cea9/.github/workflows/flutter-ci.yml#L30
```diff
-  - uses: subosito/flutter-action@4389e6cbc6cb8a4b18c628ff96ff90be0e926aa8
+  - uses: subosito/flutter-action@6a13bd08366a2aafe8605f064f9d5a42107e6f1f
```

File: .github/workflows/flutter-ci.yml [hunk 1]
```diff
@@ -30,6 +30,15 @@ jobs:
       - uses: subosito/flutter-action@4389e6cbc6cb8a4b18c628ff96ff90be0e926aa8
```
Comment (jmagman) [inline .github/workflows/flutter-ci.yml:30]:
```suggestion
      - uses: subosito/flutter-action@6a13bd08366a2aafe8605f064f9d5a42107e6f1f
```

File: .github/workflows/flutter-ci.yml [hunk 2]
```diff
@@ -30,6 +30,15 @@ jobs:
...
           channel: ${{ matrix.flutter_version }}
+      - name: Cleanup macOS
+        if: runner.os == 'macOS'
+        run: |
+          sudo rm -rf /Applications/Xcode.app \
+                      /Library/Developer/CommandLineTools
+          sudo xcode-select --reset
+      - name: Install rosetta
+        if: runner.os == 'macOS'
+        run: if [ `uname -m` = 'arm64' ]; then softwareupdate --install-rosetta; fi
```
Comment (jmagman) [inline .github/workflows/flutter-ci.yml:41]:
```suggestion
```

File: .github/workflows/flutter-ci.yml (hunk 1 above)
Comment (domesticmouse) [reply .github/workflows/flutter-ci.yml:30]: I'm currently testing subosito/flutter-action@main and it appears to be happy. I'm curious what the process for updating the pin version is. I'd also suggest bumping the versions of `actions/checkout` and `actions/setup-java`.

Example config: https://github.com/domesticmouse/codelabs/blob/master/.github/workflows/flutter-ci.yml

/cc @godofredoc

Comment (domesticmouse) [reply .github/workflows/flutter-ci.yml]: Hand applied

File: .github/workflows/flutter-ci.yml [hunk 3]
```diff
@@ -23,11 +23,12 @@ jobs:
         flutter_version: [stable, beta]
         os: [ubuntu-latest, macos-latest, windows-latest]
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-java@v1
+      - uses: actions/checkout@v3.0.1
```
Comment (godofredoc) [inline .github/workflows/flutter-ci.yml:26]: Can we use hashes here? the one for v3.0.1 is `dcd71f646680f2efd8db4afa5ad64fdcba30e748`

File: .github/workflows/flutter-ci.yml [hunk 4]
```diff
@@ -23,11 +23,12 @@ jobs:
...
+      - uses: actions/setup-java@v3.1.1
         with:
```
Comment (godofredoc) [inline .github/workflows/flutter-ci.yml:28]: Hash for this version is `9519cf1382ac8dc61ad461f7f7cb45f033220189`

**Review by godofredoc — COMMENTED** [review]
We may want to enable dependabot to automate the dependency updates in the future.

**Comment by domesticmouse** [comment]
> We may want to enable dependabot to automate the dependency updates in the future.

How do I enable dependabot to update GitHub actions?

File: .github/workflows/flutter-ci.yml (hunk 4 above)
Comment (domesticmouse) [reply .github/workflows/flutter-ci.yml:28]: done

File: .github/workflows/flutter-ci.yml (hunk 3 above)
Comment (domesticmouse) [reply .github/workflows/flutter-ci.yml]: done

**Comment by parlough** [comment]
@domesticmouse Here's how `flutter/website` does so: https://github.com/flutter/website/blob/main/.github/dependabot.yml#L14-L17

**Review by godofredoc — APPROVED** [review]

File: .github/workflows/flutter-ci.yml [hunk 5]
```diff
@@ -23,11 +23,12 @@ jobs:
         flutter_version: [stable, beta]
         os: [ubuntu-latest, macos-latest, windows-latest]
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-java@v1
+      - uses: actions/checkout@dcd71f646680f2efd8db4afa5ad64fdcba30e748
+      - uses: actions/setup-java@9519cf1382ac8dc61ad461f7f7cb45f033220189
         with:
+          distribution: 'zulu' # See https://github.com/actions/setup-java#supported-distributions
```
Comment (jmagman) [inline .github/workflows/flutter-ci.yml:29]: Did you mean to add this?

Comment (domesticmouse) [reply .github/workflows/flutter-ci.yml:29]: The `distribution` is now required for `setup-java`. There is a choice of java distributions, with `zulu` being but one. We could alternatively use `adopt-hotspot ` or `temurin` if that would be more appropriate.

**Comment by domesticmouse** [comment]
> @domesticmouse Here's how `flutter/website` does so: https://github.com/flutter/website/blob/main/.github/dependabot.yml#L14-L17

Added

File: .github/dependabot.yaml [hunk 6]
```diff
@@ -1,6 +1,10 @@
 version: 2
 enable-beta-ecosystems: true
 updates:
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: "daily"
   - package-ecosystem: "pub"
```
Comment (godofredoc) [inline .github/dependabot.yaml:8]: I'd also recommend to enable the npm ecosystem for packages-lock.json files in in_app_purchases/. It seems like those files also require multiple updates.

**Pull request merged**